### PR TITLE
Use "parent/child" naming everywhere across the codebase

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/SyncToUpstreamStatus.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package com.virtuslab.gitmachete.backend.api;
+
+public enum SyncToUpstreamStatus {
```
Comment (mkondratek) [inline backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/SyncToUpstreamStatus.java:3]: :open_mouth:

File: frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java [hunk 2]
```diff
@@ -42,29 +42,29 @@ public DemoGitMacheteRepositorySnapshot() {
             /* fork point */ null,
             /* downstreamBranches */ List.empty(),
             /* commits */ List.empty(),
-            SyncToParentStatus.MergedToParent),
+            SyncToUpstreamStatus.MergedToUpstream),
         new NonRoot(/* name */ "build-chain",
             /* customAnnotation */ "# Green edge: branch is in sync with its parent branch",
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java:47]: TBH initially I wanted to distinguish b/w backend (_upstream/downstream_) and everything else (_parent/child_)... but now I increasingly see that this distinction is more and more artificial/useless. WDYT guys about just having _parent/child_ on every level of abstraction? CC @micpiotrowski

[the remark this answers is not in the document]
Comment (mkondratek) [reply frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java:47]: I fully agree with your second thoughts. I (probably only?) see neither specific difference nor advantage b/w these names within the backend module. In general project scope, it is more intuitive (to me) to use _parent/child_ naming.

[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontendUiTableImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/DemoGitMacheteRepositorySnapshot.java:47]: Ok! renamed everywhere to parent/child then, case closed (hopefully)

File: backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/SyncToUpstreamStatus.java (hunk 1 above)
Comment (PawelLipski) [reply backendApi/src/main/java/com/virtuslab/gitmachete/backend/api/SyncToUpstreamStatus.java]: Reverted :)

File: backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToParentStatusUnitTestSuite.java [hunk 3]
```diff
@@ -37,7 +38,8 @@ public void branchAndParentPointingSameCommitAndBranchJustCreated_inSync() {
     IGitCoreLocalBranchSnapshot childBranch = createGitCoreLocalBranch(commit);
 
     // when
-    SyncToParentStatus syncToParentStatus = invokeDeriveSyncToParentStatus(childBranch, parentBranch, MISSING_FORK_POINT);
+    SyncToParentStatus syncToParentStatus = invokeDeriveSyncToParentStatus(childBranch, parentBranch,
+        MISSING_FORK_POINT);
```
Comment (micpiotrowski) [inline backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToParentStatusUnitTestSuite.java:42]: Here and in other places: this lines probably can be joined

Comment (PawelLipski) [reply backendImpl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveSyncToParentStatusUnitTestSuite.java:42]: PR #423 will fix this issue everywhere

**Pull request merged**